bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/CustomTagRegistry.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.semantics;
+
+import java.util.Collection;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.common.registry.Registry;
+
+/**
+ * {@link CustomTagRegistry} tracks all {@link CustomTag}s from different {@link CustomTagProvider}s
+ * and provides access to them.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public interface CustomTagRegistry extends Registry<CustomTag, String> {
+
+    /**
+     * This method retrieves a single custom tag from the registry.
+     *
+     * @param uid the custom tag unique identifier
+     * @return the uniquely identified custom tag
+     * @throws CustomTagNotFoundException if no custom tag matches the input
+     */
+    CustomTag getCustomTag(String uid) throws CustomTagNotFoundException;
+
+    /**
+     * This method retrieves all custom tags that are currently available in the registry
+     *
+     * @return a collection of all available custom tags
+     */
+    Collection<CustomTag> getCustomTags();
```

Comment:
What is the benefit of this over `Registry.getAll`?

Reply:
Method removed